recipe/generator.go

```diff
@@ -0,0 +1,54 @@
+package recipe
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"text/template"
+)
+
+type FromTemplateData struct {
+	FileName string                 `json:"FileName" yaml:"FileName"`
+	Data     map[string]interface{} `json:"Data" yaml:"Data"`
```

Comment:
Can we make the tags `Filename` & `Data` to be all small case in the template. would remain consistent with everything else.

Reply:
@bsushmith  we could do this, but actually there is a reason why I don't use lowercase for those two fields.

So, to unmarshal a field, it has to be public field, which should be a capital (hence FileName is the struct field name).
We could change the tag to e.g. `yaml: "filename"` so it would be `filename` in the `templatedata.yaml`, but then when you access it from the template you still have to use `FileName` because go template can only read public field from struct.

IMO it could bring more confusion if it was `filename` in `templatedata.yaml` and `FileName` when accessing it in the template.

Reply:
Makes sense. 👍